framework/src/modules/chain/logic/block/block_bft.js

```diff
@@ -0,0 +1,573 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { getAddressFromPublicKey } = require('@liskhq/lisk-cryptography');
+const { Status: TransactionStatus } = require('@liskhq/lisk-transactions');
+const _ = require('lodash');
+const crypto = require('crypto');
+const ByteBuffer = require('bytebuffer');
+const Bignum = require('../../helpers/bignum');
+const blockVersion = require('../block_version');
+const BlockReward = require('../block_reward');
+
+const { MAX_PAYLOAD_LENGTH, FEES, TRANSACTION_TYPES } = global.constants;
+const __private = {};
+let modules;
+
+/**
+ * Main Block logic.
+ *
+ * @class
+ * @memberof logic
+ * @see Parent: {@link logic}
+ * @requires bytebuffer
+ * @requires crypto
+ * @requires helpers/bignum
+ * @requires logic/block_reward
+ * @param {Object} ed
+ * @param {ZSchema} schema
+ * @param {Transaction} transaction
+ * @returns {SetImmediate} error, this
+ * @todo Add description for the params
+ */
+class BlockBFT {
```

Comment:
Why don't we just extend the basic `Block` and override the functions which have different implementation. e.g. `getBytes` and `create`